jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsResourceCache.kt

```diff
@@ -313,9 +314,9 @@ class DefaultAwsResourceCache(
         cache.clear()
     }
 
-    override fun providerRemoved(providerId: String) = clearByCredential(providerId)
+    override fun providerRemoved(identifier: ToolkitCredentialsIdentifier) = clearByCredential(identifier.id)
 
-    override fun providerModified(provider: ToolkitCredentialsProvider) = clearByCredential(provider.id)
+    override fun providerModified(identifier: ToolkitCredentialsIdentifier) = clearByCredential(identifier.id)
```

Comment:
We have to since the backing hash map is populated by a Provider, so we can:
1. Make Provider have a getIdentnfier() and return the underlyign identifier
2. Keep it this way

If we do (1),I would still keep the .displayName() and .id() to keep the code shorter...

Whats your thoughts?